query-engine/connectors/sql-query-connector/src/filter/visitor.rs

```diff
@@ -950,6 +960,9 @@ fn default_scalar_filter(
             comparable.not_equals(Expression::from(field_ref.aliased_col(alias, ctx)).all())
         }
         ScalarCondition::Search(value, _) => {
+            if !can_have_capability(ConnectorCapability::FullTextSearch) {
+                unreachable!()
+            }
```

Comment:
Can you please summarise why all these `!can_have_capability` checks are necessary in `query-engine`? Didn't we check for them already in `psl`? For Wasm Query Engine, we already assume that a given schema is already valid.

Reply:
How would optimizer know that `FullTextSearch` is not actually used for sqlite connector in `query-core` crate? In order to do so, it will somehow will need to know that:

1. Fields, corresponding to full text search can not be created in `query-structure` crate - ok, that's easy, as you mention, we have checks already.
2. At the same time in `query-core` it will somehow need to know that `search` filter would never exist for sqlite connector. And there is no way it will be able to trace that fact from (1) on it's own.